runtime/onert/core/src/compiler/TensorRegistries.h

```diff
@@ -55,6 +56,20 @@ class TensorRegistries
     }
   }
 
+  TensorRegistries(const backend::train::TrainableBackendContexts &backend_contexts, bool is_grad)
```

Comment:
`is_grad` determines whether `TensorRegistries` is for `grad` tensors or for normal tensors.